packages/react-urql/src/hooks/useQuery.ts

```diff
@@ -158,7 +158,9 @@ export interface UseQueryState<
  * };
  * ```
  */
-export type UseQueryExecute = (opts?: Partial<OperationContext>) => void;
+export type UseQueryExecute = (
+  opts?: Partial<OperationContext>
+) => Promise<unknown>;
```

Comment:
I am wondering whether this could turn into a breaking change, i.e. something complaining about not awaiting/... when doing `onClick={() => executeQuery()}`